jupyter_book_to_htmlbook/code_processing.py

```diff
@@ -111,3 +112,67 @@ def number_codeblock(pre_block, cell_number):
                 element.replace_with(element.replace('\n', f'\n{indent}'))
 
     return cell_number
+
+
+def process_code_examples(chapter):
+    """
+    Applies appropriate data types and adds titles for formal code
+    "Examples" in the text
+    """
+    examples = chapter.find_all("div", class_="tag_example")
+
+    for example_cell in examples:
+        pre_block = example_cell.find("pre")
+        example_div = pre_block.parent
+        try:
+            comments = pre_block.find_all("span", class_="c1")
+
+            # ensure comments are within the first three spans (since we
+            # expect an empty span to start based on their highlighter)
+            if not (
+                    comments[0] in pre_block.find_all("span")[0:3] and
+                    comments[1] in pre_block.find_all("span")[0:3]
+                   ):
+                logging.warning(
+                    "Missing first two line comments for uuid and title." +
+                    f"Unable to apply example formatting to {example_cell}.")
+                return example_cell
+
+            # first comment should be uuid/name; only take first word if many
+            # to preserve id attribute validation
+            example_name = comments[0].string
+            example_name = example_name.split(" ")[1]  # remove comment hash
+            comments[0].decompose()
+            # second comment should be the example title
+            example_title = comments[1].string
+            example_title = " ".join(example_title.split(" ")[1:])
+            comments[1].decompose()
+
+            logging.info("Applying example formatting to and removing" +
+                         f" first comments from: {example_cell}")
+        except IndexError:
+            logging.warning(
+                    f"Unable to apply example formatting to {example_cell}")
+            return example_cell
+
+        # remove empty space left by decomposed spans
+        # assuming they add the empty span at the beginning of block, do [0:3]
+        for element in pre_block.contents[0:3]:
+            if (
```

Comment:
Idea: If you're looking to reduce the size of this method, some sort of white space remover would be a good candidate to add as a helper method. I am guessing that similar code is used elsewhere in the package.

Reply:
Agree, though right now we've got a case where in once instance we know it's going to be a string we're removing left spacing from, whereas in other cases we've got spans and so on to deal with (which become challenging just based on the way Beautiful Soup works, or as I understand it to work at present). _When_ this comes up again, I'll see about doing a better refactor. I've started keeping a list of things that could be done better in this (e.g., using `.get("data-type")` instead of all the `try/except` catches... ah, the things we learn as we do this more...